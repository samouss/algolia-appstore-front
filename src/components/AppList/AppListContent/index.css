.AppListContent {
  flex: 1;
  border: 1px solid #f2f2f2;
  background-color: #ffffff;
}
