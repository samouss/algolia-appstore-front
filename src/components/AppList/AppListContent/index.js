import React from 'react';
import PropTypes from 'prop-types';
import InfiniteList from 'react-infinite-scroll-list';
import AppItem, { HitPropTypes } from 'components/AppItem';
import './index.css';

const AppListContent = ({
  hits,
  isLoading,
  isEndReached,
  onNextPage,
}) => (
  <InfiniteList
    styleName="AppListContent"
    containerTagName="ul"
    isLoading={isLoading}
    isEndReached={isEndReached}
    isAttachOnWindow
    onReachThreshold={onNextPage}
    threshold={680}
  >
    {hits.map(hit => (
      <AppItem
        key={hit.objectID}
        name={hit.name}
        image={hit.image}
        link={hit.link}
        rating={hit.rating}
        ratingCount={hit.ratingCount}
        price={hit.price}
        _highlightResult={hit._highlightResult}
      />
    ))}
  </InfiniteList>
);

AppListContent.propTypes = {
  hits: PropTypes.arrayOf(PropTypes.shape(HitPropTypes)).isRequired,
  isLoading: PropTypes.bool.isRequired,
  isEndReached: PropTypes.bool.isRequired,
  onNextPage: PropTypes.func.isRequired,
};

export default AppListContent;
