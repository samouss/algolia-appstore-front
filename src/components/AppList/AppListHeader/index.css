.AppListHeader {
  min-height: 11px;
  margin-bottom: 4px;

  @media (--tablet) {
    min-height: 15px;
  }
}

.AppListHeader__Stat {
  font-size: 0.75rem;
  text-align: right;
  font-style: italic;
  color: #b9b9b9;

  @media (--tablet) {
    font-size: 0.85rem;
  }
}

.AppListHeader__StatHits {
  font-weight: 500;
  font-size: 0.85rem;

  @media (--tablet) {
    font-size: 0.95rem;
  }
}
