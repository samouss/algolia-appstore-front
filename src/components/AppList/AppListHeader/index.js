import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const AppList = ({
  nbHits,
  processingTimeMS,
  isInitialLoad,
}) => (
  <div styleName="AppListHeader">
    {!isInitialLoad && (
      <p styleName="AppListHeader__Stat">
        <span styleName="AppListHeader__StatHits">{nbHits.toLocaleString()} results </span>
        <span>found in </span>
        <span>{processingTimeMS}ms</span>
      </p>
    )}
  </div>
);

AppList.propTypes = {
  nbHits: PropTypes.number.isRequired,
  processingTimeMS: PropTypes.number.isRequired,
  isInitialLoad: PropTypes.bool.isRequired,
};

export default AppList;
