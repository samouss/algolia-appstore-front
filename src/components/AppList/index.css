.AppList {
  width: 90vw;
  /* @TODO: move margin to Layout component */
  margin: 2.5vh auto;
}

.AppList__Header {
  min-height: 11px;
  margin-bottom: 4px;

  @media (--tablet) {
    min-height: 15px;
  }
}

.AppList__Stat {
  font-size: 0.65rem;
  text-align: right;
  font-style: italic;
  color: #b9b9b9;

  @media (--tablet) {
    font-size: 0.85rem;
  }
}

.AppList__StatHits {
  font-weight: 500;
}

.AppList__StatTime {

}

.AppList__Content {
  border: 1px solid #f2f2f2;
  background-color: #ffffff;
  min-height: 95vh;
}
