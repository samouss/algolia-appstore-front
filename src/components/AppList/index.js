import React from 'react';
import PropTypes from 'prop-types';
import AppItem, { HitPropTypes } from 'components/AppItem';
import './index.css';

// @TODO: maybe split this component for
// <HeaderList>
// <HeaderContent>
// something like that, will see

const AppList = ({
  hits,
  nbHits,
  processingTimeMS,
  isInitialLoad,
}) => (
  <div styleName="AppList">
    <div styleName="AppList__Header">
      {!isInitialLoad && (
        <p styleName="AppList__Stat">
          <span styleName="AppList__StatHits">{nbHits.toLocaleString()} results </span>
          <span>found in </span>
          <span styleName="AppList__StatTime">{processingTimeMS}ms</span>
        </p>
      )}
    </div>

    <ul styleName="AppList__Content">
      {hits.map(hit => (
        <AppItem
          key={hit.objectID}
          name={hit.name}
          image={hit.image}
          link={hit.link}
          rating={hit.rating}
          ratingCount={hit.ratingCount}
          price={hit.price}
          _highlightResult={hit._highlightResult}
        />
      ))}
    </ul>
  </div>
);

AppList.propTypes = {
  hits: PropTypes.arrayOf(PropTypes.shape(HitPropTypes)).isRequired,
  nbHits: PropTypes.number.isRequired,
  processingTimeMS: PropTypes.number.isRequired,
  isInitialLoad: PropTypes.bool.isRequired,
};

export default AppList;
