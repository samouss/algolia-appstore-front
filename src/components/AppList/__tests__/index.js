import React from 'react';
import { shallow } from 'enzyme';
import AppList from '../index';

describe('<AppList />', () => {
  const defaultProps = {
    hits: [
      {
        objectID: '1234',
        name: 'Cut the Rope',
        image: 'http://image.com',
        link: 'http://link.com',
        rating: 3,
        ratingCount: 1234,
        price: '0 $',
        _highlightResult: {
          name: {
            value: '__algolia__highlight__pre__tag__Cut the__algolia__highlight__post__tag__ Rope',
          },
        },
      },
      {
        objectID: '5678',
        name: 'Angry Birds',
        image: 'http://image.com',
        link: 'http://link.com',
        rating: 5,
        ratingCount: 6789,
        price: '0 $',
        _highlightResult: {
          name: {
            value: 'Angry Birds',
          },
        },
      },
    ],
    nbHits: 500,
    processingTimeMS: 5,
    isInitialLoad: false,
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <AppList
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render on initial load', () => {
    const props = {
      ...defaultProps,
      isInitialLoad: true,
    };

    const component = shallow(
      <AppList
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
