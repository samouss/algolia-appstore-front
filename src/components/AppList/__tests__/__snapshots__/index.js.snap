// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppList /> expect to render 1`] = `
<div
  styleName="AppList"
>
  <div
    styleName="AppList__Header"
  >
    <p
      styleName="AppList__Stat"
    >
      <span
        styleName="AppList__StatHits"
      >
        500
         results 
      </span>
      <span>
        found in 
      </span>
      <span
        styleName="AppList__StatTime"
      >
        5
        ms
      </span>
    </p>
  </div>
  <ul
    styleName="AppList__Content"
  >
    <AppItem
      _highlightResult={
        Object {
          "name": Object {
            "value": "__algolia__highlight__pre__tag__Cut the__algolia__highlight__post__tag__ Rope",
          },
        }
      }
      image="http://image.com"
      link="http://link.com"
      name="Cut the Rope"
      price="0 $"
      rating={3}
      ratingCount={1234}
    />
    <AppItem
      _highlightResult={
        Object {
          "name": Object {
            "value": "Angry Birds",
          },
        }
      }
      image="http://image.com"
      link="http://link.com"
      name="Angry Birds"
      price="0 $"
      rating={5}
      ratingCount={6789}
    />
  </ul>
</div>
`;

exports[`<AppList /> expect to render on initial load 1`] = `
<div
  styleName="AppList"
>
  <div
    styleName="AppList__Header"
  />
  <ul
    styleName="AppList__Content"
  >
    <AppItem
      _highlightResult={
        Object {
          "name": Object {
            "value": "__algolia__highlight__pre__tag__Cut the__algolia__highlight__post__tag__ Rope",
          },
        }
      }
      image="http://image.com"
      link="http://link.com"
      name="Cut the Rope"
      price="0 $"
      rating={3}
      ratingCount={1234}
    />
    <AppItem
      _highlightResult={
        Object {
          "name": Object {
            "value": "Angry Birds",
          },
        }
      }
      image="http://image.com"
      link="http://link.com"
      name="Angry Birds"
      price="0 $"
      rating={5}
      ratingCount={6789}
    />
  </ul>
</div>
`;
